conductor_api/src/conductor/passphrase_manager.rs

```diff
@@ -0,0 +1,110 @@
+use crossbeam_channel::{unbounded, Sender};
+use holochain_core_types::error::error::HolochainError;
+use holochain_sodium::secbuf::SecBuf;
+use std::{
+    sync::{Arc, Mutex},
+    thread,
+    time::{Duration, Instant},
+};
+
+/// We are caching the passphrase for 10 minutes.
+const PASSPHRASE_CACHE_DURATION_SECS: u64 = 600;
+
+pub trait PassphraseService {
+    fn request_passphrase(&self) -> Result<SecBuf, HolochainError>;
+}
+
+#[derive(Clone)]
+pub struct PassphraseManager {
+    passphrase_cache: Arc<Mutex<Option<SecBuf>>>,
+    passphrase_service: Arc<Mutex<PassphraseService + Send>>,
+    last_read: Arc<Mutex<Instant>>,
+    timeout_kill_switch: Sender<()>,
+}
+
+impl PassphraseManager {
+    pub fn new(passphrase_service: Arc<Mutex<PassphraseService + Send>>) -> Self {
+        let (kill_switch_tx, kill_switch_rx) = unbounded::<()>();
+        let pm = PassphraseManager {
+            passphrase_cache: Arc::new(Mutex::new(None)),
+            passphrase_service,
+            last_read: Arc::new(Mutex::new(Instant::now())),
+            timeout_kill_switch: kill_switch_tx,
+        };
+
+        let pm_clone = pm.clone();
+
+        thread::spawn(move || loop {
```

Comment:
I think we're going to need to do an audit of system threads at some point... they should probably mostly be moved into some kind of pool executor scheme.

Reply:
:+1: